db/translistener.c

```diff
@@ -1166,22 +1166,31 @@ int javasp_load_procedure_int(const char *name, const char *param,
     }
 
     p = malloc(sizeof(struct stored_proc));
+    if(!p) {
+        logmsg(LOGMSG_ERROR, "OOM %s\n", __func__);
+        rc = -1;
+        goto done;
+    }
     p->name = strdup(name);
-    if (paramvalue)
+    if(!p->name) {
+oom:
+        logmsg(LOGMSG_ERROR, "OOM %s\n", __func__);
+        free(p);
+        rc = -1;
+        goto done;
+    }
+    if (paramvalue) {
         p->qname = strdup(name);
+        if(!p->qname) goto oom;
+    }
     if (param)
         p->param = strdup(param);
     else
         p->param = strdup("<sc>");
+    if(!p->param) goto oom;
```

Comment:
Does not really matter, but we are not freeing p->qname if we goto oom from here. Nor p->name either.